packages/cli/src/sso.ee/saml/saml.service.ee.ts

```diff
@@ -236,20 +241,58 @@ export class SamlService {
 		};
 	}
 
-	async setSamlPreferences(prefs: Partial<SamlPreferences>): Promise<SamlPreferences | undefined> {
+	async setSamlPreferences(
+		prefs: Partial<SamlPreferences>,
+		tryFallback: boolean = false,
+	): Promise<SamlPreferences | undefined> {
 		await this.loadSamlify();
 		await this.loadPreferencesWithoutValidation(prefs);
 		if (prefs.metadataUrl) {
-			const fetchedMetadata = await this.fetchMetadataFromUrl();
-			if (fetchedMetadata) {
-				this._samlPreferences.metadata = fetchedMetadata;
+			try {
+				const fetchedMetadata = await this.fetchMetadataFromUrl();
+				if (fetchedMetadata) {
+					this._samlPreferences.metadata = fetchedMetadata;
+				} else {
+					// in this case the metadata url didn't produce a valid metadata for SAML
+					// therefore we are rejecting the change to it
+					throw new InvalidSamlMetadataUrlError();
```

Comment:
This is not covered by a test as far as I can tell.

![image](https://github.com/user-attachments/assets/a35030cd-1edf-4322-b8a3-5af6ff214272)